__init__.py

```diff
@@ -11,1810 +11,1070 @@
-
-        if limit is not None:
-            q["cnt"] = limit
-        data = self.request({
-            "path": "/forecast/daily",
-            "query": q
-        })
-        return self.to_forecast(data, "daily")
-
-    def to_forecast(self, data, interval):
-        forecast = self.forecast.parse_JSON(data)
-        if forecast is not None:
-            forecast.set_interval(interval)
-            return Forecaster(forecast)
-        else:
-            return None
-
-    def set_OWM_language(self, lang):
-        self.owmlang = lang
-
-        # Certain OWM condition information is encoded using non-utf8
-        # encodings. If another language needs similar solution add them to the
-        # encodings dictionary
-        encodings = {
-            'se': 'latin1'
-        }
-        self.encoding = encodings.get(lang, 'utf8')
+from mycroft import MycroftSkill, intent_handler
+from mycroft.messagebus.message import Message
+from mycroft.util.parse import extract_number
+from .skill import (
+    CurrentDialog,
+    DAILY,
+    DailyDialog,
+    DailyWeather,
+    HOURLY,
+    HourlyDialog,
+    get_dialog_for_timeframe,
+    LocationNotFoundError,
+    OpenWeatherMapApi,
+    WeatherConfig,
+    WeatherDialog,
+    WeatherIntent,
+    WeatherReport,
+    WeeklyDialog,
+)
+
+# TODO: VK Failures
+#   Locations: Washington, D.C.
+#
+# TODO: Intent failures
+#   Later weather: only the word "later" in the vocab file works all others
+#       invoke datetime skill
+
+MARK_II = "mycroft_mark_2"
+TWELVE_HOUR = "half"
+CLEAR = 0
+PARTLY_CLOUDY = 1
+CLOUDY = 2
+LIGHT_RAIN = 3
+RAIN = 4
+THUNDERSTORM = 5
+SNOW = 6
+WINDY = 7
 
 
 class WeatherSkill(MycroftSkill):
+    """Main skill code for the weather skill."""
+
     def __init__(self):
         super().__init__("WeatherSkill")
-
-        # Build a dictionary to translate OWM weather-conditions
-        # codes into the Mycroft weather icon codes
-        # (see https://openweathermap.org/weather-conditions)
-        self.CODES = multi_key_dict()
-        self.CODES['01d', '01n'] = 0                # clear
-        self.CODES['02d', '02n', '03d', '03n'] = 1  # partly cloudy
-        self.CODES['04d', '04n'] = 2                # cloudy
-        self.CODES['09d', '09n'] = 3                # light rain
-        self.CODES['10d', '10n'] = 4                # raining
-        self.CODES['11d', '11n'] = 5                # stormy
-        self.CODES['13d', '13n'] = 6                # snowing
-        self.CODES['50d', '50n'] = 7                # windy/misty
+        self.weather_api = OpenWeatherMapApi()
+        self.weather_api.set_language_parameter(self.lang)
+        self.weather_config = WeatherConfig(self.config_core, self.settings)
+        self.platform = self.config_core["enclosure"].get("platform", "unknown")
 
         # Use Mycroft proxy if no private key provided
         self.settings["api_key"] = None
         self.settings["use_proxy"] = True
```

Comment:
I don't like code that alludes to future functionality.  If we don't support it now, it is gone!

Reply:
this used to be there in the first versions, i always assumed it was accidentally removed (not even removed, it just broke at some point with some untested change that made it to the skill)

are there no plans to allow users to provide their own keys? 

since a refactor is ongoing, and its trivial to add this, why not add it now instead of adding a TODO or removing it for good?

i consider this a very important feature, as will a lot of privacy minded users

Reply:
Yeah it's functionality that we do want to add. In terms of this PR, it's intention was to be a refactor rather than adding functionality, but then it's changed significantly with the API update etc. So could do it here or in a follow up PR. 

That mycroft-core readme definitely needs an update either way as it would be retrieving the key from the settings rather than mycroft.conf.

The PR from Ulno never got finished off unfortunately. I think a big part of that is because the Skill needed some tough love like it's just received. This is a much nicer base to be working from.